### PR TITLE
feat(dashboard): add feature flags provider and hook

### PR DESCRIPTION
**Pull request by ChmaraX**
### What changed? Why was the change needed?

Adds LD provider and hooks to be able to use feature flags.

Fixes: NV-4901

### CODE REVIEW

File: apps/dashboard/src/context/feature-flags-provider.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  };
+});
+
+export function FeatureFlagsProvider({ children }: { children: React.ReactNode }) {
+  return (
+    <Suspense fallback={<div>Loading...</div>}>
+      <AsyncFeatureFlagsProvider>{children}</AsyncFeatureFlagsProvider>
+    </Suspense>
+  );
+}
```
Comment (ChmaraX) [inline apps/dashboard/src/context/feature-flags-provider.tsx:38]: I think this is a bit more elegant and logically separated compared to what [LD docs suggests](https://docs.launchdarkly.com/sdk/client-side/react/react-web#initialize-the-client):

```jsx
(async () => {
  const FeatureFlagsProvider = await asyncWithLDProvider({
    ....
  });

  createRoot(document.getElementById('root')!).render(
    <StrictMode>
      <FeatureFlagsProvider>
        <RouterProvider router={router} />
      </FeatureFlagsProvider>
    </StrictMode>
  );
})();
```

Comment (SokratisVidros) [reply apps/dashboard/src/context/feature-flags-provider.tsx:38]: I like the logic with Suspense but please make sure that there is no initial flickering involved between a render where LD flags are not loaded yet and the time they arrive. See how we did it in [the web dashboard](https://github.com/novuhq/novu/blob/a305ec690cbdd7b89e636379cbfd36657d55a662/apps/web/src/index.tsx#L17) where lazy was harder to work.

File: apps/dashboard/src/hooks/use-feature-flag.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  return !!LAUNCH_DARKLY_CLIENT_SIDE_ID;
+}
+
+export const useFeatureFlag = (key: FeatureFlagsKeysEnum, defaultValue = false): boolean => {
+  const flags = useFlags();
+
+  if (!isLaunchDarklyEnabled()) {
+    const envValue =
+      // Check window._env_ first
+      window?._env_?.[key] ??
```
Comment (SokratisVidros) [inline apps/dashboard/src/hooks/use-feature-flag.tsx]: I don't think we have the `_env_` fallback on the window object in the new Dashboard. This was in the old web app. Instead, we should rely on VITE_ prefixed env variables as a fallback.

**Review by SokratisVidros — CHANGES_REQUESTED** [review]

File: apps/dashboard/src/context/feature-flags-provider.tsx (hunk 1 above)
Comment (ChmaraX) [reply apps/dashboard/src/context/feature-flags-provider.tsx:38]: I can't reproduce any flickering - it waits until the `AsyncFeatureFlagsProvider` resolves to either LD or to empty wrapper, only after that the app is rendered, should be exactly as in the legacy web.

```tsx
const AsyncFeatureFlagsProvider = lazy(async () => {
  if (!LAUNCH_DARKLY_CLIENT_SIDE_ID) {
    return {
      default: ({ children }: { children: React.ReactNode }) => <>{children}</>,
    };
  }

  const LaunchDarklyProvider = await asyncWithLDProvider(LD_CONFIG);
  return {
    default: ({ children }: { children: React.ReactNode }) => <LaunchDarklyProvider>{children}</LaunchDarklyProvider>,
  };
});

export function FeatureFlagsProvider({ children }: { children: React.ReactNode }) {
  return (
    <Suspense>
      <AsyncFeatureFlagsProvider>{children}</AsyncFeatureFlagsProvider>
    </Suspense>
  );
}
```

Comment (SokratisVidros) [reply apps/dashboard/src/context/feature-flags-provider.tsx:38]: Nice. Do we need a loader or a splash screen? PTAL at [this touch](https://github.com/novuhq/novu/blob/4408daa2c02c938efa4ca71a6e5e9fa6ede69f03/apps/web/src/ApplicationReadyGuard.tsx#L20). It fades in the app when loading of everything is done. It's pretty cool.

**Review by SokratisVidros — APPROVED** [review]

File: apps/dashboard/src/context/feature-flags-provider.tsx (hunk 1 above)
Comment (ChmaraX) [reply apps/dashboard/src/context/feature-flags-provider.tsx:38]: The LD initiation is very fast - we didn't had splash screen in the legacy web before, so it should be probably ok.

**Pull request merged**